### ISSUE TITLE
proposal: bare variable use in condition should raise linting error

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/ansible-lint#668]
Because the code below would not work as expected even if you are using perfectly valid YAML booleans, we need a new rule to tell user to add the `| bool` shenanigan. 

```yaml
- var:
    foo: true
  when: foo  # <-- this needs `| bool` to make it work correctly. 
  debug:
    msg: be surprused that I do not run....
```

**Comment by webknjaz** [comment]
In this case, you need to ignore literals like `False`
